### ISSUE TITLE
Build issue with Webpack/Typescript/Babel

### ISSUE THREAD

**Issue by joustava** [ultraq/icu-message-formatter#14]
We are seeing an issue during our build when trying to use version 0.14.2 of this library, where the build errors with
**Module not found: Error: Default condition should be last one**
[Reproduced in this repo](https://github.com/joustava/reproduce-issue)

```
yarn build
yarn run v1.22.21
$ yarn build:clean && webpack --node-env=production
$ rimraf build; mkdir -p build
assets by status 149 bytes [cached] 1 asset
./src/index.ts 111 bytes [built] [code generated]

ERROR in ./src/index.ts 1:0-5:39
Module not found: Error: Default condition should be last one

webpack 5.90.3 compiled with 1 error in 165 ms
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

A possible fix is to change the order of the 'default' key in the package.json such that it looks like:
```
"exports": {
		"import": {
			"types": "./dist/icu-message-formatter.d.ts",
			"default": "./dist/icu-message-formatter.js"
		},
		"require": {
			"types": "./dist/icu-message-formatter.d.cts",
			"default": "./dist/icu-message-formatter.cjs"
		}
	},
```

The same needs to be done in the @ultraq/function-utils repo and then (at least for us) the build is succesfull again.

**Comment by ultraq** [comment]
Ah yep, makes sense.  I haven't seen the error before with my webpack project, but I'm probably just running a different version.  Thanks for the PRs - if they're all good I'll try get a new version of these out this weekend.

**Comment by joustava** [comment]
We were seeing these errors during a project update so it's probably due to using newer versions. I removed the `yarn. lock` files in both PR's. If anything else needs to be done on these let me know!

**Comment by ultraq** [comment]
I've just released icu-message-formatter 0.14.3 which contains the fix from both repos - thanks again for submitting these 🙌
